src/Internal/Payment/State/VerifyState.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Class VerifyState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WC_Payments_API_Payment_Intention;
+use WCPay\Constants\Payment_Intent_Status;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Internal\Service\OrderService;
+use WCPay\Internal\Service\PaymentRequestService;
+
+/**
+ * This state is used to create payment intent and verify next actions based on the intent state.
+ */
+class VerifyState extends AbstractPaymentState {
+
+	/**
+	 * Order service.
+	 *
+	 * @var OrderService
+	 */
+	private $order_service;
+
+	/**
+	 * Payment request service.
+	 *
+	 * @var PaymentRequestService
+	 */
+	private $payment_request_service;
+
+	/**
+	 * Class constructor, only meant for storing dependencies.
+	 *
+	 * @param StateFactory          $state_factory           Factory for payment states.
+	 * @param OrderService          $order_service           Service for order-related actions.
+	 * @param PaymentRequestService $payment_request_service Connection with the server.
+	 */
+	public function __construct(
+		StateFactory $state_factory,
+		OrderService $order_service,
+		PaymentRequestService $payment_request_service
+	) {
+		parent::__construct( $state_factory );
+
+		$this->order_service           = $order_service;
+		$this->payment_request_service = $payment_request_service;
+	}
+
+	/**
+	 * Process all needed verifications.
+	 *
+	 * @return AbstractPaymentState
+	 * @throws \WCPay\Internal\Payment\Exception\StateTransitionException
+	 */
+	public function process() {
+		$context = $this->get_context();
+
+		// Payments are currently based on intents, request one from the API.
+		try {
+			$intent = $this->payment_request_service->create_intent( $context );
+
+			// Something went wrong since the intent is not an instance of payment intent class. Transition to error state.
+			if ( ! $intent instanceof WC_Payments_API_Payment_Intention ) {
```

Comment:
`create_intent` should either return an intent, or throw an exception. This part should not be needed.